sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/Readme.md

```diff
@@ -37,23 +44,101 @@ var client = new TimeSeriesInsightsClient(
     tokenCredential);
 ```
 
-## Create, get, replace, and delete instances
+# Time Series Insights Samples
+
+The following section provides several code snippets using the `client` created above, and covers the main functions of Time Series Insights. You can explore and learn more about the Time Series Insights client library APIs through using the samples project.
+
+- [Time Series Insights ID](#time-series-insights-id)
+- [Time Series Insights Model Settings](#time-series-insights-model-settings)
+- [Time Series Insights Instances](#time-series-insights-instances)
+- [Time Series Insights Types](#time-series-insights-types)
+- [Time Series Insights Hierarchies](#time-series-insights-hierarchies)
+- [Time Series Insights Query](#time-series-insights-query)
 
-### Create instances
-You can use the following snippet to create instance Id using `TimeSeriesId`.
+## Time Series Insights ID
+A single Time Series ID value is composed of up to 3 string values that uniquely identify a Time Series instance. The keys that make up the Time Series ID are chosen when creating a Time Series Insights Gen2 environment through the Azure portal. The position of values must match Time Series ID properties specified on the environment and returned by Get Model Setting API.  For example, if your Time Series Insights environment is setup with ID properties `Building`, `Floor` and `Room`, then this code snippet illustrates creating a Time Series instance ID using `TimeSeriesId` class for building 'Millennium', 'Floor2' and room '2A01'. Visit [this page][tsi_id_learn_more] to check out the best practices for choosing a Time Series ID.
 
 ```csharp
-you can use the Model Settings client to learn more about what the Time Series Id is composed of in the environment.
-TimeSeriesId instanceId = modelSettings.TimeSeriesIdProperties.Count switch
+var instanceId =  new TimeSeriesId("Millennium", "Floor2", "2A01");
+```
+
+## Time Series Insights Model Settings
+
+Use [ModelSettings](https://github.com/Azure/azure-sdk-for-net/blob/82c7aff5ef867f7a3d77f5ac43a0b1c5576996e6/sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs#L32) in the `TimeSeriesInsightsClient` to learn more about the environment model settings, such as name, default type ID, and the properties that define the Time Series ID during environment creation.
```

Comment:
link is not from master branch.

Reply:
Yes this is a perm link. That means, even if master changed, this reference does not change. I believe it references a specific commit that will always remain.

Reply:
I see the concern so if there is any code modification this document can be out of date!
But also if we change examples, this will be pointing to old code then.

Reply:
Good point. Let me take a look at how other SDKs do it.

Reply:
For the one SDKs I checked out so far, all use master. I will update the links to point to master. Good catch.